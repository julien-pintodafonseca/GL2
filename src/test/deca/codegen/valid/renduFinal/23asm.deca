// Description:
//      Ajout d'une méthode en langage assembleur avec affichage d'une chaine de caractère
//
// Resultats:
//      A:OK
//      3.5;

class A {
    float x = 3.5;
    float getX() asm("WSTR \"OK\"
        WNL
        LOAD -2(LB), R1
        LOAD 1(R1), R0
        RTS");
}

{
    A oA = new A();
    println("A:", oA.getX(), ";");
}
