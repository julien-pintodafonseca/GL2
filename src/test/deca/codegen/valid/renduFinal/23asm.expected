A:OK
3.50000e+00;
